graphbolt/src/cuda/common_hip.h

```diff
@@ -0,0 +1,242 @@
+/**
```

Comment:
Oof this is a fair amount of duplication

Reply:
Totally fair, this was a sledgehammer to get things working. I'll take a look at this tomorrow and cut down on the duplicate code